scratch/evaluate_gaming_traffic.cc

```diff
@@ -62,38 +117,46 @@ main (int argc, char *argv[])
 
   Ipv4InterfaceContainer interfaces = address.Assign (devices);
 
-  // Packet inter-arrival time
-  Ptr<LogNormalRandomVariable> iat_first = CreateObjectWithAttributes<LogNormalRandomVariable> ("Mu", DoubleValue (0.34),
-                                                                                                "Sigma", DoubleValue (1.73));
-  Ptr<ConstantRandomVariable> iat_second = CreateObjectWithAttributes<ConstantRandomVariable> ("Constant", DoubleValue (100));
-
-  Ptr<MixtureRandomVariable> iat_dist = CreateObject<MixtureRandomVariable> ();
-  iat_dist->SetRandomVariables ({iat_first, iat_second}, {0.5725, 0.4275});
-
-
-  // Packet size
-  Ptr<UniformRandomVariable> pkt_first = CreateObjectWithAttributes<UniformRandomVariable> ("Min", DoubleValue (0),
-                                                                                            "Max", DoubleValue (1355));
-  Ptr<ConstantRandomVariable> pkt_second = CreateObjectWithAttributes<ConstantRandomVariable> ("Constant", DoubleValue (1356));
-
-  Ptr<MixtureRandomVariable> pkt_dist = CreateObject<MixtureRandomVariable> ();
-  pkt_dist->SetRandomVariables ({pkt_first, pkt_second}, {0.3606, 0.6394});
-
-  GamingStreamingServerHelper gamingStreamingHelper ("ns3::GamingStreamingServer", interfaces.GetAddress (0), 9);
-  gamingStreamingHelper.SetAttribute ("InterArrivalTime", PointerValue (iat_dist));
-  gamingStreamingHelper.SetAttribute ("PacketSize", PointerValue (pkt_dist));
+  GamingStreamingServerHelper gamingStreamingHelper (gameId, interfaces.GetAddress (0), 9);
 
   ApplicationContainer serverApps = gamingStreamingHelper.Install (nodes.Get (1));
-  serverApps.Start (Seconds (1.0));
-  serverApps.Stop (Seconds (3.0));
+  gamingServer = StaticCast<GamingStreamingServer> (serverApps.Get (0));
+  serverApps.Start (Seconds (0.1));
+  serverApps.Stop (Seconds (simulationTime + 0.1));
   /** End of Streaming Server*/
-  PacketSinkHelper sink ("ns3::TcpSocketFactory", InetSocketAddress (Ipv4Address::GetAny (), 9));
+
+  PacketSinkHelper sink ("ns3::UdpSocketFactory", InetSocketAddress (Ipv4Address::GetAny (), 9));
   ApplicationContainer sinkApps = sink.Install (nodes.Get (0));
+  packetSink = StaticCast<PacketSink> (sinkApps.Get (0));
   sinkApps.Start (Seconds (0.0));
-  sinkApps.Stop (Seconds (3.1));
-  Simulator::Stop (Seconds (3.1));
+  sinkApps.Stop (Seconds (simulationTime + 0.2));
```

Comment:
why +0.2?

Reply:
I want to be sure that the sink receives all the packets sent by the server before the simulation end.
I'm not sure about the clean way to do this.

Reply:
when a user manually inputs `simulationTime`, it expects the simulation to end at `simulationTime`. It is not necessary that the sink receives every single packet anyway, we're just collecting stats here